lib/io/buffer.toit

```diff
@@ -22,6 +22,9 @@ class Buffer extends CloseableWriter:
   buffer_/ByteArray := ?
   is-growable_/bool := ?
 
+  processed -> int:
+    return offset_
```

Comment:
We don't update the processed_ field at all, it's not just when clearing.

Reply:
You don't need to update the `processed_` field otherwise, as the base class is doing it for you.
But by clearing, the buffer class is somehow influencing the "processed" number. You could argue that it shouldn't reset, but it's probably better that it does.
-> If we changed the `processed_` to 0 in the super class, then things should work.
Maybe that's actually better, since the base class could then, in theory, use that field for other things.